packages/core/src/entity/ArrayCollection.ts

```diff
@@ -177,6 +177,32 @@ export class ArrayCollection<T extends object, O extends object> {
     return this.items.has(entity);
   }
 
+
+  /**
+   * Extracts a slice of elements starting at position start to end (exclusive) of the Collection.
+   * If end is null it returns all elements from $start to the end of the Collection.
+   */
+
+  slice(start = 0, end?: number): T[]  {
+    let counter = 0;               // counter to keep track of the index of the collection
+    end = end ?? this.items.size; // if end is undefined, set it to the size of the collection
+    const items = new Set<T>();
+
+    for (const item of this.items) {
+
+    if (counter === end) { break; } // break the loop if we reach the end of the slice
+
+    if (counter >= start &&  counter < end) {
```

Comment:
```suggestion
    if (counter >= start && counter < end) {
```

we really need to make this fail in eslint, i am sure there is a rule for this